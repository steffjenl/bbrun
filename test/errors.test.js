import shell from 'shelljs';
function run(options) {
  return shell.exec(`node index.js ${options}`, {
    silent: true
  });
}

describe("should fail properly when script section is mispelled", () => {
  it("running all steps in a pipeline", () => {
    const result = run(
      "--template test/templates/mispelled-scripts.yml --pipeline branches:master --dry-run"
    );
    expect(result.stdout).toBe("");
    expect(result.stderr).toMatchSnapshot();
    expect(result.code).toBe(1);
  });

  it("running a step by name", () => {
    const result = run(
      "run_dev --template test/templates/mispelled-scripts.yml --pipeline branches:dev --dry-run"
    );
    expect(result.stdout).toBe("");
    expect(result.stderr).toMatchSnapshot();
    expect(result.code).toBe(1);
  });
});

it("non-existing template should fail", () => {
  const res = run("--template foo.yml --dry-run");
  expect(res.stdout).toBe("");
  expect(res.stderr).toContain("foo.yml can't be found\n");
  expect(res.code).toBe(1);
});

it("invalid yaml should fail", () => {
  const res = run("--template test/templates/empty-template.yml --dry-run");
  expect(res.stdout).toBe("");
  expect(res.stderr).toMatchSnapshot();
  expect(res.code).toBe(1);
});

it("empty template file should fail", () => {
  const res = run("--template test/templates/empty-template.yml --dry-run");
  expect(res.stdout).toBe("");
  expect(res.stderr).toMatchSnapshot();
  expect(res.code).toBe(1);
});
