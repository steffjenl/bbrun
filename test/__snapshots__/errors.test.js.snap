// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty template file should fail 1`] = `
"file:///Users/stephan/Projects/bbrun/src/template.js:68
    throw new Error(
          ^

Error: Cannot read properties of undefined (reading 'pipelines')
Malformed template, check https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html
    at validate (file:///Users/stephan/Projects/bbrun/src/template.js:68:11)
    at new Template (file:///Users/stephan/Projects/bbrun/src/template.js:103:5)
    at readTemplate (file:///Users/stephan/Projects/bbrun/src/template.js:17:10)
    at default (file:///Users/stephan/Projects/bbrun/src/bbrun.js:13:20)
    at file:///Users/stephan/Projects/bbrun/index.js:86:3
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.13.0
"
`;

exports[`invalid yaml should fail 1`] = `
"file:///Users/stephan/Projects/bbrun/src/template.js:68
    throw new Error(
          ^

Error: Cannot read properties of undefined (reading 'pipelines')
Malformed template, check https://confluence.atlassian.com/bitbucket/configure-bitbucket-pipelines-yml-792298910.html
    at validate (file:///Users/stephan/Projects/bbrun/src/template.js:68:11)
    at new Template (file:///Users/stephan/Projects/bbrun/src/template.js:103:5)
    at readTemplate (file:///Users/stephan/Projects/bbrun/src/template.js:17:10)
    at default (file:///Users/stephan/Projects/bbrun/src/bbrun.js:13:20)
    at file:///Users/stephan/Projects/bbrun/index.js:86:3
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.13.0
"
`;

exports[`should fail properly when script section is mispelled running a step by name 1`] = `
"\\"script\\" section not found in step:
{
    \\"name\\": \\"run_dev\\",
    \\"scripts\\": [
        \\"echo dev\\"
    ]
}
"
`;

exports[`should fail properly when script section is mispelled running all steps in a pipeline 1`] = `
"\\"script\\" section not found in step:
{
    \\"scripts\\": [
        \\"echo master\\"
    ]
}
"
`;
